x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/TransportRollupIndexerAction.java

```diff
@@ -170,9 +179,12 @@ protected RollupIndexerAction.Response newResponse(
     private class Async extends AsyncBroadcastAction {
         private final RollupIndexerAction.Request request;
         private final ActionListener<RollupIndexerAction.Response> listener;
+        private final Task task;
+        private boolean hasCancelled = false;
```

Comment:
I don't think I fully understand that the changes to the `Async` inner class.
It seems to always cancel a shard level rollup before actually performing it?

Reply:
I think I understand now, why this logic is added and how it works. This for cancelling other shard level operations in case one fails. Downsampling may take a long time to complete and if a downsample fails in one shard, this can speed up the response to the client.

Reply:
Should the  `hasCancelled` field be volatile? I think it will read / updated from different threads.

Reply:
done

Reply:
I also don't see this change. Maybe you forgot to push it?